### PR TITLE
media/audio_manager : apply pthread_unlock when missing case

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
1. Before return, apply pthread_unlock

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f07f99bd941bcf460b2df9c8c90f9560f096b996] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f07f99bd941bcf460b2df9c8c90f9560f096b996] - Code Rule Check (C++) OK.

File: framework/src/media/audio/audio_manager.c [hunk 1]
```diff
@@ -640,16 +646,20 @@ audio_manager_result_t set_audio_stream_out(unsigned int channels, unsigned int
...
-			pcm_close(g_audio_out_cards[g_actual_audio_out_card_id].pcm);
-			return AUDIO_MANAGER_RESAMPLE_FAIL;
+			ret = AUDIO_MANAGER_RESAMPLE_FAIL;
+			goto error_with_pcm;
 		}
 		medvdbg("resampler ratio = %f, buffer_size = %d\t", g_audio_out_cards[g_actual_audio_out_card_id].resample.ratio, g_audio_out_cards[g_actual_audio_out_card_id].resample.buffer_size);
 		medvdbg("buffer address = 0x%x\n", g_audio_out_cards[g_actual_audio_out_card_id].resample.buffer);
 	}
-
+
```
Comment (pradeepns85) [inline framework/src/media/audio/audio_manager.c]: remove extra space

File: framework/src/media/audio/audio_manager.c [hunk 2]
```diff
@@ -576,20 +577,25 @@ audio_manager_result_t set_audio_stream_in(unsigned int channels, unsigned int s
...
 	}
 
 	pthread_mutex_unlock(&(g_audio_in_cards[g_actual_audio_in_card_id].card_mutex));
-
-	return AUDIO_MANAGER_SUCCESS;
+	return ret;
+	
+error_with_pcm:
+	pcm_close(g_audio_in_cards[g_actual_audio_in_card_id].pcm);
+	pthread_mutex_unlock(&(g_audio_in_cards[g_actual_audio_in_card_id].card_mutex));
```
Comment (pradeepns85) [inline framework/src/media/audio/audio_manager.c]: I see that pthread_mutex_lock/ unlock has been performed to protect the global resource  g_audio_in_cards , But i have a concern here, What if the context switch happens after the line no 540 and then somebody modifies the g_actual_audio_in_card_id to INVALID as that global variable is not protected ?  
Isn't it the rest of the logic after line no 40 leads to wrong result ?  What is your opinion?

File: framework/src/media/audio/audio_manager.c (hunk 1 above)
Comment (Taejun-Kwon) [reply framework/src/media/audio/audio_manager.c]: done

File: framework/src/media/audio/audio_manager.c (hunk 2 above)
Comment (Taejun-Kwon) [reply framework/src/media/audio/audio_manager.c]: yes I understand what you concerning, But actually g_actual_audio_in_card_id is kindly temp code, will fix later.

**Comment by seinfra** [comment]
Target : [9e930478f86711de295f27795b943c3d1ebede67] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9e930478f86711de295f27795b943c3d1ebede67] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7bb38e69875408b6960a52ed8340eae6fca21701] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7bb38e69875408b6960a52ed8340eae6fca21701] - Code Rule Check OK.

**Pull request merged**